gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/KafkaJobStatusMonitor.java

```diff
@@ -275,6 +279,7 @@ static void addJobStatusToStateStore(org.apache.gobblin.configuration.State jobS
       modifyStateIfRetryRequired(jobStatus);
```

Comment:
post-`ReevaluateDagProc`, can't we keep the reading of and deciding about retries minimal in or even absent from this class and situate it over there?

Reply:
yes, but how to tell the reevaluate dag proc, which job status to consider for reading?
does it make sense or i did not get the question correctly?